process/hyperOutportBlocksQueue.go

```diff
@@ -0,0 +1,63 @@
+package process
+
+import (
+	"errors"
+	"sync"
+
+	data "github.com/multiversx/mx-chain-ws-connector-template-go/data/hyperOutportBlocks"
+)
+
+type hyperOutportBlocksQueue struct {
+	mu       sync.RWMutex
+	Elements []*data.HyperOutportBlock
+	Size     int
+}
+
+func NewHyperOutportBlocksQueue() *hyperOutportBlocksQueue {
+	return &hyperOutportBlocksQueue{
+		Elements: make([]*data.HyperOutportBlock, 0),
+	}
+}
+
+func (q *hyperOutportBlocksQueue) Enqueue(elem *data.HyperOutportBlock) error {
+	q.mu.Lock()
+	defer q.mu.Unlock()
+
+	// TODO: add queue size to config
+	//if q.GetLength() == q.Size {
+	//	return errors.New("hyperOutportBlocksQueue is full")
+	//}
+	q.Elements = append(q.Elements, elem)
+	return nil
+}
+
+func (q *hyperOutportBlocksQueue) Dequeue() (*data.HyperOutportBlock, error) {
+	q.mu.Lock()
+	defer q.mu.Unlock()
```

Comment:
I think this will need to be reworked if all the exported methods will have their own locking and unlocking of the mutex.

Reply:
changed to buffered channel.